### PR TITLE
BSD: Add support

### PR DESCRIPTION
**Pull request by Er2ch**

### CODE REVIEW

File: engine/sys_dll.cpp [hunk 1]
```diff
@@ -456,15 +459,17 @@ void Sys_Error_Internal( bool bMinidump, const char *error, va_list argsList )
...
+#elif defined( OSX ) || defined(BSD)
 		// Doing this doesn't quite work the way we want because there is no "crashing" thread
 		// and we see "No thread was identified as the cause of the crash; No signature could be created because we do not know which thread crashed" on the back end
 		//SteamAPI_WriteMiniDump( 0, NULL, build_number() );
 		printf("\n ##### Sys_Error: %s", text );
 		fflush(stdout );
 
+#ifndef BSD
 		int *p = 0;
 		*p = 0xdeadbeef;
```
Comment (nillerusr) [inline engine/sys_dll.cpp:471]: use 
`raise( SIGTRAP );`

File: vgui2/vgui_surfacelib/linuxfont.cpp [hunk 2]
```diff
@@ -20,7 +20,12 @@
 #include <tier0/dbg.h>
 #include <vgui/ISurface.h>
 #include <utlbuffer.h>
-#include <fontconfig/fontconfig.h>
+#ifdef __FreeBSD__
+  // why not -I/usr/local/include? this breaks other part of code
+# include "/usr/local/include/fontconfig/fontconfig.h"
```
Comment (nillerusr) [inline vgui2/vgui_surfacelib/linuxfont.cpp:25]: What the fuck?

File: vgui2/vgui_surfacelib/linuxfont.cpp [hunk 3]
```diff
@@ -32,6 +37,11 @@
 
 #define FT_LOAD_FLAGS	0 //$ (FT_LOAD_TARGET_LIGHT)
 
+#ifdef BSD
+#undef HAVE_FC
+#define HAVE_FC 0
```
Comment (nillerusr) [inline vgui2/vgui_surfacelib/linuxfont.cpp:42]: Remove this lines

File: appframework/sdlmgr.cpp [hunk 4]
```diff
@@ -129,7 +129,7 @@ static void DebugPrintf( const char *pMsg, ... )
 	Plat_DebugString( buf );
 }
 
-// #define SDLAPP_DEBUG
```
Comment (nillerusr) [inline appframework/sdlmgr.cpp:132]: bruh

File: appframework/glmdisplaydb_linuxwin.inl [hunk 5]
```diff
@@ -632,7 +632,7 @@ void    GLMDisplayInfo::Dump( int which )
         SDLAPP_FUNC;
 
         GLMPRINTF(("\n         #%d: GLMDisplayInfo @ %08x, pixwidth=%d  pixheight=%d",
-                           which, (int)this,  m_info.m_displayPixelWidth,  m_info.m_displayPixelHeight ));
+                           which, (int)(size_t)this,  m_info.m_displayPixelWidth,  m_info.m_displayPixelHeight ));
```
Comment (nillerusr) [inline appframework/glmdisplaydb_linuxwin.inl:635]: Use this example:
`printf("%p\n", this);`

**Comment by Er2ch** [comment]
some TODOs:

- [ ] Test on non-FreeBSD systems (Open and maybe NetBSD)

- [ ] Fix no labels in main menu (gameui/BasePanel.cpp)

- [ ] Fix debug builds (SDL crashes when init GL library, maybe because of amdgpu)

- [ ] Fix bug when chapter names in first runs (fixes after some restarts) shows as something like #rtal/Chapter_1_Name

**Pull request merged**